src/server/req_message.c

```diff
@@ -341,7 +342,18 @@ req_relnodesjob(struct batch_request *preq)
 	if ((nodeslist != NULL) && (nodeslist[0] == '\0')) {
 		nodeslist = NULL;
 	}
-	rc = free_sister_vnodes(pjob, nodeslist, msg, LOG_BUF_SIZE, preq);
+
+	if (preq->rq_extend && *preq->rq_extend) {
+		keep_select = strchr(preq->rq_extend, '=');
+		if (!keep_select || !*(keep_select+1))
+			rc = PBSE_INVALSELECTRESC;
+		if (rc || (rc = validate_perm_res_in_select(++keep_select, 1))) {
```

Comment:
What you are doing here is not wrong, but it might be more readable as...
```
		if (keep_select && *(keep_select+1))
			keep_select++;
		else
			rc = PBSE_INVALSELECTRESC;
		if (rc || (rc = validate_perm_res_in_select(keep_select, 1))) {
```

Reply:
Agree, added it in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a